### PR TITLE
initial setup

### PR DESCRIPTION
**Pull request by simonrondelez**

### CODE REVIEW

File: rds/main.tf [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  default = {
+    mysql = "3306"
+    oracle = "1521"
+  }
+}
+
+variable "families" {
+  default = {
+    mysql = "mysql5.6"
+    oracle = "oracle-se2-12.1"
```
Comment (MattiasGees) [inline rds/main.tf]: No love for postgres?

File: rds/main.tf [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+resource "aws_db_subnet_group" "rds" {
+  name = "${var.project}-${var.environment}${var.tag}-rds"
+  description = "Our main group of subnets"
+  subnet_ids = ["${var.subnets}"]
+}
+
+resource "aws_db_parameter_group" "rds_mysql" {
+  name = "mysql-rds-${var.project}-${var.environment}${var.tag}"
```
Comment (SamClinckspoor) [inline rds/main.tf]: I would leave parameter groups out of the module, as these vary a lot depending on the project.

File: rds/main.tf [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  family = "${lookup(var.families, "mysql")}"
+  description = "rds ${var.project} ${var.environment} parameter group for mysql"
+  parameter = { name="slow_query_log" value="1" }
```
Comment (MattiasGees) [inline rds/main.tf]: Will these work with other databases than mysql?

File: rds/main.tf [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  }
+
+  tags {
+    Name = "${var.project}-${var.environment}${var.tag}-sg_rds"
+    Environment = "${var.environment}"
+    Project = "${var.project}"
+  }
+}
+
+resource "aws_db_subnet_group" "rds" {
```
Comment (SamClinckspoor) [inline rds/main.tf:34]: isn't this already created in the vpc module?

**Review by SamClinckspoor — COMMENTED** [review]
a few comments. can you also run `terraform fmt`?

File: rds/main.tf (hunk 1 above)
Comment (simonrondelez) [reply rds/main.tf]: https://github.com/skyscrapers/terraform-rds/pull/1/commits/e1e5ca05761d3b5d2b1969b8080dbd6521df75bc

File: rds/main.tf (hunk 3 above)
Comment (simonrondelez) [reply rds/main.tf]: its a seperate aws_db_parameter_group per type of DB

File: rds/main.tf (hunk 2 above)
Comment (simonrondelez) [reply rds/main.tf]: what if we use those as default but add the option to override it in the RDS resource?

Comment (SamClinckspoor) [reply rds/main.tf]: ok for me

File: rds/main.tf [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  parameter_group_name    = "${var.rds_type}-rds-${var.project}-${var.environment}${var.tag}"
+  multi_az                = "${var.multi_az}"
+  replicate_source_db     = "${var.replicate_source_db}"
+  backup_retention_period = "${var.backup_retention_period}"
+  storage_encrypted       = "${var.storage_encrypted}"
+  apply_immediately       = "${var.apply_immediately}"
+  skip_final_snapshot     = false
+
+  tags {
+    Name        = "${var.project}-${var.environment}${var.tag}-rds01"
```
Comment (MattiasGees) [inline rds/main.tf]: the number is hardcoded

File: rds/main.tf [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+resource "aws_db_parameter_group" "rds_postgres" {
+  name        = "postgres-rds-${var.project}-${var.environment}${var.tag}"
+  family      = "${lookup(var.families, "postgres")}"
+  description = "rds ${var.project} ${var.environment} parameter group for postgres"
+}
+
+resource "aws_db_instance" "rds" {
+  identifier              = "${var.project}-${var.environment}${var.tag}-rds01"
```
Comment (MattiasGees) [inline rds/main.tf]: The number is hard coded

File: rds/main.tf [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  allocated_storage       = "${var.storage}"
+  engine                  = "${lookup(var.engines, var.rds_type)}"
+  engine_version          = "${lookup(var.engine_versions, var.rds_type)}"
+  instance_class          = "${var.size}"
+  storage_type            = "${var.storage_type}"
+  username                = "root"
+  password                = "${var.rds_password}"
+  vpc_security_group_ids  = ["${aws_security_group.sg_rds.id}"]
+  db_subnet_group_name    = "${aws_db_subnet_group.rds.id}"
+  parameter_group_name    = "${var.rds_type}-rds-${var.project}-${var.environment}${var.tag}"
```
Comment (SamClinckspoor) [inline rds/main.tf]: this is not variable, the exact name will always be the same as the parametergroups you create in this module

**Review by SamClinckspoor — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+* [`replicate_source_db`]: String(optional) RDS source to replicate from
+* [`multi_az`]: bool(optional) Multi AZ for RDS master (default: true)
+* [`backup_retention_period`]: int(optional) How long do you want to keep RDS backups (default: 14)
+* [`apply_immediately`]: bool(optional) whether you want to Apply changes immediately (default: true)
+* [`storage_encrypted`]: bool(optional) whether you want to Encrypt RDS storage (default: true)
+* [`tag`]: String(optional) tag
+* [`project`]: String(required) the name of the project this RDS belongs to
+* [`environment`]: String(required) the name of the environment these subnets belong to (prod,stag,dev)
+* [`number`]: int(optional) number of the database (default 01)
+* [`rds_parameter_group_name`]: String(required) the parameter group that is used for the db
```
Comment (MattiasGees) [inline README.md]: Maybe add a list of the choices that we add by default?

Comment (simonrondelez) [reply README.md]: added

**Review by MattiasGees — APPROVED** [review]

**Pull request merged**